### PR TITLE
fix: parse the process name to get the process type instead of using port count

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/7393

Get the process type by parsing the name instead of using port count.

### CODE REVIEW

**Comment by ChanYiLin** [comment]
Need to wait until IM PR merged, so I can update the vendor here.

File: engineapi/instance_manager.go
Comment (innobead) [inline engineapi/instance_manager.go:326]: Suggest using a regular expression to check instead. We can explicitly understand what the pattern is.

File: engineapi/instance_manager.go
Comment (innobead) [inline engineapi/instance_manager.go:320]: Can we have a `longhorn.InstanceNone` to replace the empty string here?

File: engineapi/instance_manager.go
Comment (innobead) [inline engineapi/instance_manager.go:334]: Can just call `getTypeForInstance(longhorn.InstanceNone, name)` in this function to prevent duplicated code.

**Review by innobead — DISMISSED** [review]

File: engineapi/instance_manager.go
Comment (ChanYiLin) [reply engineapi/instance_manager.go]: Fixed, thanks

File: engineapi/instance_manager.go
Comment (ChanYiLin) [reply engineapi/instance_manager.go:326]: Fixed, thanks

File: engineapi/instance_manager.go
Comment (ChanYiLin) [reply engineapi/instance_manager.go:334]: Fixed, thanks

**Comment by ChanYiLin** [comment]
Wait for https://github.com/longhorn/go-common-libs/pull/25

**Comment by innobead** [comment]
@ChanYiLin any updates?

**Comment by ChanYiLin** [comment]
@innobead
wait for https://github.com/longhorn/longhorn-instance-manager/pull/407

**Comment by ChanYiLin** [comment]
> This pull request is now in conflict. Could you fix it @ChanYiLin? 🙏

fixed.

**Comment by ChanYiLin** [comment]
> @ChanYiLin any updates?

Waiting for https://github.com/longhorn/longhorn-instance-manager/pull/407
So we can update the vendor in manager

File: engineapi/instance_manager.go
Comment (derekbit) [inline engineapi/instance_manager.go:18]:
```
import (
	"context"
	"fmt"
	"path/filepath"
	"strconv"

	"github.com/pkg/errors"
	"github.com/sirupsen/logrus"
	"go.uber.org/multierr"

	imapi "github.com/longhorn/longhorn-instance-manager/pkg/api"
	imclient "github.com/longhorn/longhorn-instance-manager/pkg/client"
	immeta "github.com/longhorn/longhorn-instance-manager/pkg/meta"
	imutil "github.com/longhorn/longhorn-instance-manager/pkg/util"
	lhutils "github.com/longhorn/go-common-libs/utils"

	"github.com/longhorn/longhorn-manager/datastore"
	"github.com/longhorn/longhorn-manager/types"

	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
)
```

Comment (ChanYiLin) [reply engineapi/instance_manager.go]: @derekbit 
Fixed, thanks!
BTW, can you briefly explain what is our current import placement order?
Or is there any reference I can check, thanks

Comment (derekbit) [reply engineapi/instance_manager.go]: I learnt it from previous commits as well.

```
import (
	// golang standard library packages such as "fmt", "path/filepath" and so on.
	"context"
	"fmt"
	"path/filepath"
	"strconv"

        // third-party packages not related to k8s
	"github.com/pkg/errors"
	"github.com/sirupsen/logrus"
	"go.uber.org/multierr"

        // third-party packages without alias names related to k8s
        corev1 "k8s.io/api/core/v1"

        // third-party packages with alias names related to k8s
        "k8s.io/kubernetes/pkg/controller"

        // packages without alias names from longhorn components except for this repo 

        // packages with alias names from longhorn components except for this repo 
	imapi "github.com/longhorn/longhorn-instance-manager/pkg/api"
	imclient "github.com/longhorn/longhorn-instance-manager/pkg/client"
	immeta "github.com/longhorn/longhorn-instance-manager/pkg/meta"
	imutil "github.com/longhorn/longhorn-instance-manager/pkg/util"
	lhutils "github.com/longhorn/go-common-libs/utils"

        // packages without alias names from this repo
	"github.com/longhorn/longhorn-manager/datastore"
	"github.com/longhorn/longhorn-manager/types"

        // packages with alias names from this repo
        longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
)
```

**Review by derekbit — APPROVED** [review]

File: engineapi/instance_manager.go
Comment (ChanYiLin) [reply engineapi/instance_manager.go]: Thanks!

**Pull request merged**
